rfi_file_monitor/utils/decorators.py

```diff
@@ -133,14 +133,16 @@ def wrapper(self: Operation, file: File):
 
                 # run the wrapped method, and do the usual exception and return value handling
                 try:
-                    rv = run(self, _file)
+                    metadata, rv = run(self, _file)
                 except SkippedOperation:
                     continue
                 # other exceptions should propagate
 
+                if _file.operation_metadata:
+                    file.operation_metadata[_file.filename] = _file.operation_metadata.values()
```

Comment:
`operation_metadata` is a dict with keys the numbers of the operation indices that attached data.

So you will need to do something like:

```python
file.operation_metadata[self.index][_file.filename] = _file.operation_metadata[self.index]
```

Ensure that before the loop over the files starts, that a dict is assigned to `file.operation_metadata[self.index]`.